src/components/MeasurementPlot.tsx

```diff
@@ -0,0 +1,66 @@
+import { xyToXYObject } from 'ml-spectra-processing';
+import React from 'react';
+import { Axis, Heading, LineSeries, Plot } from 'react-plot';
+
+interface Variable {
+  label: string;
+  data: number[];
+  units?: string;
+  symbol?: string;
+}
+interface Data {
+  variables: Record<string, Variable>;
+}
+interface Measurement {
+  meta?: object;
+  filename?: string;
+  info?: object;
+  title?: string;
+  data: Data[];
+}
+export interface MeasurementPlotProps {
+  measurment: Measurement;
+  dataIndex?: number;
+  xVariableName?: string;
+  yVariableName?: string;
+  width?: number;
+  height?: number;
+}
+export function MeasurementPlot(props: MeasurementPlotProps) {
+  const {
+    measurment,
+    dataIndex = 0,
+    xVariableName = 'x',
+    yVariableName = 'y',
+    width = 800,
+    height = 400,
+  } = props;
+  const { title = '', data } = measurment;
+  return (
+    <Plot width={width} height={height}>
+      <Heading title={title} />
+      {data.map(
+        ({ variables: { [xVariableName]: x, [yVariableName]: y } }, i) => [
+          <LineSeries
+            // eslint-disable-next-line react/no-array-index-key
+            key={i}
+            data={xyToXYObject({
+              x: x.data.splice(dataIndex),
```

Comment:
This is not what we would like here. The dataIndex is expected to select the data at the top level (sorry I used 'data' twice in the object hierarchy).

```
  "title": "Mattson Instruments",
  "data": [
    {
````

In the new example data is an array of 2 elements and it should therefore work with dataIndex 0 and 1.

On a side node please take care about the difference between `splice` and `slice`. Splice is 'in-place' meaning you actually change the original array and therefore there is a side effect we nearly never want. `slice` will make a copy (will take memory and could be slow).

Also if we need to send part of an array to a function and the array is a typed array we can use `subarray`. This will not be a copy but just a new pointer to the part of the typed array. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/subarray

Reply:
fixed,
thank you